src/models/utils/index.ts

```diff
@@ -19,3 +22,129 @@ export function onlyHasFields(obj: object, fields: string[]): boolean {
 export function isFlagEnabled(Flags: number, checkFlag: number): boolean {
   return (checkFlag & Flags) === checkFlag;
 }
+
+/**
+ * Sets a transaction's flags to its numeric representation
+ *
+ * @param {Transaction} tx A transaction to set its flags to its numeric representation
+ * @returns {void}
+ */
+export function setTransactionFlagsToNumber(tx: Transaction): void {
+  switch (tx.TransactionType) {
+    case 'OfferCreate':
+      tx.Flags = convertOfferCreateFlagsToNumber(tx.Flags as OfferCreateFlags);
+      break;
+    case 'PaymentChannelClaim':
+      tx.Flags = convertPaymentChannelClaimFlagsToNumber(tx.Flags as PaymentChannelClaimFlags);
+      break;
+    case 'Payment':
+      tx.Flags = convertPaymentTransactionFlagsToNumber(tx.Flags as PaymentTransactionFlags);
+      break;
+    case 'TrustSet':
+      tx.Flags = convertTrustSetFlagsToNumber(tx.Flags as TrustSetFlags);
+      break;
+    default:
+      tx.Flags = convertGlobalFlagsToNumber(tx.Flags);
```

Comment:
`convertGlobalFlagsToNumber` should be applied on all transactions, regardless of their type

Reply:
It technically is in my PR but looks like it's not apparent reading it. I call `convertGlobalFlagsToNumber` in each `convert{TransactionType}FlagsToNumber` so that these functions return the full numeric value including GlobalFlags.

If we still want to keep `GlobalFlags` interface, I can move `convertGlobalFlagsToNumber` to the top of `setTransactionFlagsToNumber` and specify `convert{TransactionType}FlagsToNumber` exclude `GlobalFlags`.

Reply:
It technically is in my PR but looks like it's not apparent reading it. I call `convertGlobalFlagsToNumber` in each `convert{TransactionType}FlagsToNumber` so that these functions return the full numeric value including `GlobalFlags`.

If we still want to keep `GlobalFlags` interface, I can move `convertGlobalFlagsToNumber` to the top of `setTransactionFlagsToNumber` and specify `convert{TransactionType}FlagsToNumber` exclude `GlobalFlags`.

Reply:
If we still want to keep `GlobalFlags` interface, I can move `convertGlobalFlagsToNumber` to the top of `setTransactionFlagsToNumber` and specify `convert{TransactionType}FlagsToNumber` exclude `GlobalFlags`.

Reply:
I don't quite understand what you mean

Reply:
I just refactored it, it should be clear now. 🙂